zhaquirks/tuya/thermostat_88teujp.py

```diff
@@ -0,0 +1,251 @@
+"""Saswell (Tuya whitelabel) 88teujp thermostat valve quirk."""
+
+from functools import reduce
+import logging
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomDevice
+from zigpy.zcl import foundation
+from zigpy.zcl.clusters.general import Basic, Identify, Ota, PowerConfiguration
+from zigpy.zcl.clusters.hvac import Thermostat
+
+from . import TUYA_GET_DATA, TUYA_SET_DATA, TUYA_SET_DATA_RESPONSE, TuyaManufCluster
+from .. import Bus, LocalDataCluster
+from ..const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+OCCUPIED_HEATING_SETPOINT_REPORTED = "occupied_heating_setpoint_reported"
+SYSTEM_MODE_REPORTED = "system_mode_reported"
+LOCAL_TEMP_REPORTED = "local_temp_reported"
+BATTERY_REPORTED = "battery_reported"
+
+OCCUPIED_HEATING_SETPOINT_COMMAND_ID = 615
+SYSTEM_MODE_COMMAND_ID = 357
+LOCAL_TEMP_COMMAND_ID = 614
+BATTERY_STATE_COMMAND_ID = 1385
+
+OCCUPIED_HEATING_SETPOINT_ATTR = 0x0012
+SYSTEM_MODE_ATTR = 0x001C
+RUNNING_MODE_ATTR = 0x001E
+LOCAL_TEMP_ATTR = 0x0000
+CTRL_SEQ_OF_OPER_ATTR = 0x001B
+BATTERY_PERCENTAGE_REMAINING_ATTR = 0x0021
+
+MIN_HEAT_SETPOINT_ATTR = 0x0015
+MAX_HEAT_SETPOINT_ATTR = 0x0016
+
+
+def payload_to_decimal(data):
+    """Coverts command payload to a single decimal.
+
+    [4, 0, 0, 1, 39] => 295 and [4, 0, 0, 0, 220] => 220
+    """
+    return reduce(lambda acc, i: ((acc << 8) + i) % 2 ** 32, data[1:], 0)
+
+
+def decimal_to_payload(number):
+    """Coverts decimal to command payload.
+
+    e.g. 295 => [4, 0, 0, 1, 39] and 220 => [4, 0, 0, 0, 220]
+    """
+    hex = "{:X}".format(number).rjust(4, "0")
+    chunk1 = int(hex[0:2], 16)
+    chunk2 = int(hex[2:], 16)
+    return [4, 0, 0, chunk1, chunk2]
+
+
+class ManufacturerThermostatCluster(TuyaManufCluster):
+    """Manufacturer thermostat cluster."""
+
+    def handle_cluster_request(self, tsn, command_id, args):
```

Comment:
Again, reusing `TuyaManufacturerClusterAttributes` enables acting as if the device was using standard zigbee attributes to communicate, which makes the rest of the code simpler.